Introduce ADR (#1042)
owncloud/ocis: docs/adr/0002-persist-accounts-using-cs3-storage.md

# 2. Persist accounts in a CS3 storage

* Status: accepted <!-- optional -->
* Deciders: @butonic, @felixboehm <!-- optional -->
* Date: 2020-08-21 <!-- optional -->

Technical Story: [File system based indexing](https://github.com/owncloud/ocis-accounts/pull/92) <!-- optional -->

## Context and Problem Statement

> To set up HA or a geo replicated setup we need to persist accounts in a distributed way. Furthermore, the bleve index makes the accounts service stateful, which we wand to avoid for a scale out deployment.

Reviewer: ```suggestion
To set up High Availability (HA) or a geo-replicated setup we need to persist accounts in a distributed way. Furthermore, the [bleve](https://github.com/blevesearch/bleve) index makes the accounts service stateful, which we want to avoid for a scale out deployment.
```

> To set up HA or a geo replicated setup we need to persist accounts in a distributed way. Furthermore, the bleve index makes the accounts service stateful, which we wand to avoid for a scale out deployment.

Reviewer 2: ```suggestion
To set up HA or a geo replicated setup we need to persist accounts in a distributed way. Furthermore, the bleve index makes the accounts service stateful, which we want to avoid for a scale out deployment.
```

> 

Reviewer 3: The last sentence might be more clear like that: _In addition to the accounts data a bleve index should be persisted as well for scale out deployments._

Author: That is not what I meant. I'll rephrase.

## Considered Options

* Look into distributed bleve
* Persist users in a CS3 storage

## Decision Outcome

> Chosen option: "Persist users in a CS3 storage", because we have one service less running and can rely on the filesystem for geo replication and HA.

Reviewer: ```suggestion
Chosen option: "Persist users in a CS3 storage", because we have one service less running and can rely on the filesystem for geo-replication and HA.
```
